### DISCUSSION TITLE
Where are the `http-statuses` translation keys are actually used in Laravel?

### DISCUSSION THREAD

**Discussion by jhm-ciberman** [Laravel-Lang/common#80] [Q&A] [answered]
Question: Where are the `http-statuses` translation keys are actually used in Laravel? 
I can't find any reference to that in the [Laravel Framework](https://github.com/laravel/framework) repository source code nor in the laravel docs.

**Comment by andrey-helldar** [comment, marked as answer]
The keys for this project are not used under the hood of the Laravel framework.

In turn, Laravel uses a direct translation of phrases such as [`Not Found`](https://github.com/laravel/framework/blob/10.x/src/Illuminate/Foundation/Exceptions/views/404.blade.php) or [`Page Expired`](https://github.com/laravel/framework/blob/10.x/src/Illuminate/Foundation/Exceptions/views/419.blade.php).

The [`HTTP Statuses`](https://github.com/Laravel-Lang/http-statuses) package allows developers to use these keys themselves in their project.

For example:
```php
namespace App\Exceptions;

class Handler extends BaseHandler
{
    public function register(): void
    {
        $this->registerDefault();
    }

    protected function registerDefault(): void
    {
        $this->renderable(function (Throwable $e) {
            $code = method_exists($e, 'getStatusCode') ? $e->getStatusCode() : $e->getCode();

            return config('app.debug')
                ? $this->response($e->getMessage(), (int) $code)
                : $this->response(__('http-statuses.' . $code), (int) $code);
        });
    }
}
```

**Reply by jhm-ciberman** [reply]
Oh. I got it! 😁

Is there any way (config file, CLI switch) to disable the generation and upgrade of those files?
We are not using the http-status codes nor the custom attributes in validations ("validations.attributes"). 
We don't want to polute our translations with fields we don't use.

**Reply by andrey-helldar** [reply]
Yes, sure. To do this, follow these steps:

```bash
composer remove laravel-lang/common

composer require --dev laravel-lang/publisher laravel-lang/lang
```

You can read about it in the [package](https://laravel-lang.com/packages-http-statuses.html) page of our documentation.

**Reply by jhm-ciberman** [reply]
You rock! Thank you.
And thanks for this amazing package!

**Reply by andrey-helldar** [reply]
You're welcome!

**Reply by buddhaCode** [reply]
@andrey-helldar Can you re-add this approach to the docs? The "Alternative" section isn't present anymore.

**Reply by andrey-helldar** [reply]
@buddhaCode, we've optimized the look and feel of the menu by removing unnecessary attachments, as well as separating the Installation section. This information is now located on each package page in the "Packages" section.

In the case of the "HTTP Statuses" package, it can be found here: https://laravel-lang.com/packages-http-statuses.html

**Reply by buddhaCode** [reply]
@andrey-helldar I am a new user of this project. I am just looking for a translation for the validation messages. I don't need localised routes, models or HTTP status codes.
As a new user, I first see the "Basic Usage" chapter. In it I should install the common package. However, this simply contains all packages. I may not even need many of them. I only see that the project is divided into several packages two main chapters later under "Packages".
I find the start and the introduction a bit bumpy if I really only need a few features and not everything.

**Reply by andrey-helldar** [reply]
@buddhaCode The "Common" package will indeed install all of our packages. In fact, that's what it was designed for - to simplify the installation and upgrade process.

In your case, you only need the "Lang" package - it contains both validation messages and covers other installed Laravel packages in case they are installed in your application.

Information on it can be found here: https://laravel-lang.com/packages-lang.html

As for the Basic Usage section, I will take your concern into consideration and think about how it can be improved.

**Reply by buddhaCode** [reply]
Thank you.
